### PR TITLE
[MOB-9422] makes tryMergeUser private

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s) if any

* [MOB-9422](https://iterable.atlassian.net/browse/MOB-9422)

## ✏️ Description

> This pull request makes the tryMergeUser function private and removes source email from parameters.

[MOB-9422]: https://iterable.atlassian.net/browse/MOB-9422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserMerge.java [hunk 1]
```diff
@@ -3,12 +3,12 @@
 public class AnonymousUserMerge {
     private static final String TAG = "AnonymousUserMerge";
 
-    public void tryMergeUser(IterableApiClient apiClient, String sourceUserId, String sourceEmail, String destinationUserIdOrEmail, boolean isEmail, boolean merge, boolean shouldUserDefaultMerge, MergeResultCallback callback) {
+    void tryMergeUser(IterableApiClient apiClient, String sourceUserId, String destinationUserIdOrEmail, boolean isEmail, boolean merge, boolean shouldUserDefaultMerge, MergeResultCallback callback) {
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/AnonymousUserMerge.java:6]: `sourceUserId` should also be named `sourceUserIdOrEmail` here to reduce confusion

**Review by Ayyanchira — COMMENTED** [review]
Renaming of parameters required.

**Review by evantk91 — COMMENTED** [review]
Addressed comment and removed some unnecessary tests

**Review by Ayyanchira — APPROVED** [review]
LGTM

**Comment by evantk91** [comment]
> LGTM

Re

> LGTM

Requesting re-review. I added additional checks to the setUserId and setEmail. Originally the checks were in the attemptAndProcessMerge function but I think these should not be called if user is being logged out, if the anonymous user is being set or if the user is switching known profiles.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 2]
```diff
@@ -873,8 +861,8 @@ private void setUserId(@Nullable String userId, @Nullable String authToken, bool
         onLogin(authToken);
     }
 
-    private void attemptAndProcessMerge(String userIdOrEmail, boolean isEmail, boolean merge, boolean shouldUseDefaultMerge, IterableHelper.FailureHandler failureHandler, String sourceUserId, String sourceEmail) {
-        anonymousUserMerge.tryMergeUser(apiClient, sourceUserId, sourceEmail, userIdOrEmail, isEmail, merge, shouldUseDefaultMerge, (mergeResult, error) -> {
+    private void attemptAndProcessMerge(@NonNull String userIdOrEmail, boolean isEmail, boolean merge, boolean shouldUseDefaultMerge, IterableHelper.FailureHandler failureHandler, String sourceUserId) {
+        anonymousUserMerge.tryMergeUser(apiClient, sourceUserId, userIdOrEmail, isEmail, merge, shouldUseDefaultMerge, (mergeResult, error) -> {
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:865]: source -> anonymousUserId
userIdOrEmail -> destinationUserIdOREmail / destinationUser

**Review by Ayyanchira — APPROVED** [review]
Made some nit pick comments :)

**Pull request merged**